src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1088,7 +1088,7 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
 
             if (state.CancellationToken.IsCancellationRequested)
             {
-                state.Tcs.TrySetCanceled();
+                state.Tcs.TrySetCanceled(state.CancellationToken);
```

Comment:
There's a call to TrySetCanceled inside of the Register callback; presumably that should be updated as well, e.g.:
```C#
var cancellationTokenRegistration = state.CancellationToken.Register(s =>
{
    RequestState rs = (RequestState)s;
    rs.Tcs.TrySetCanceled(rs.CancellationToken);
}, state);
```
?